pkg/client/client.go

```diff
@@ -638,6 +643,87 @@ func newConfiguration(ctrlRuntimeClient client.Client, secretName, namespace str
 	return config, nil
 }
 
+func fetchCredentialsIniFromSecret(secret *corev1.Secret, config *providers.Configuration) error {
+	creds, ok := secret.Data["credentials"]
+	if !ok {
+		return fmt.Errorf("failed to fetch key 'credentials' in secret data")
+	}
+	inifile, err := ini.Load(creds)
+	if err != nil {
+		return err
+	}
+	s, err := inifile.GetSection(kubeIniDefaultSection)
+	if err != nil {
+		return err
+	}
+	if s.HasKey(kubeCredentialTypeString) {
+		accessType, err := s.GetKey(kubeCredentialTypeString)
+		if err != nil {
+			return err
+		}
+		switch at := accessType.String(); at {
+		// access key
+		case kubeCredentialAccessKey:
+			if s.HasKey(kubeAccessKeyID) && s.HasKey(kubeAccessKeySecret) {
+				id, err := s.GetKey(kubeAccessKeyID)
+				if err != nil {
+					return err
+				}
+				secret, err := s.GetKey(kubeAccessKeySecret)
+				if err != nil {
+					return err
+				}
+				config.AccessKeyID = id.String()
+				config.AccessKeySecret = secret.String()
+			}
```

Comment:
Potentially move this and the other case logic into helpers to make this function easier to read